### PR TITLE
Implement waiting in the rollout service

### PR DESCRIPTION
**Pull request by hannesg**
The frontend part will come in a second MR. The configurable limit will also be implemented there

### CODE REVIEW

File: services/rollout-service/pkg/service/broadcast.go [hunk 1]
```diff
@@ -192,29 +196,85 @@ func (b *Broadcast) StreamStatus(req *api.StreamStatusRequest, svc api.RolloutSe
...
-			if s.RolloutStatus == api.RolloutStatus_RolloutStatusSuccesful {
-				continue
+		s := filterApplication(req, r)
+		if s != nil {
+			apps[r.Key] = s
+		}
+	}
+	status := aggregateStatus(apps)
+	if wait != nil {
+		// The waiting function is used in testing to make sure, we are really processing delayed events.
```
Comment (sven-urbanski-freiheit-com) [inline services/rollout-service/pkg/service/broadcast.go:214]: The comment makes it sound a bit like waiting is only used in testing? Is that accurate? Maybe rephrase.

File: services/rollout-service/pkg/service/broadcast.go [hunk 2]
```diff
@@ -192,29 +196,85 @@ func (b *Broadcast) StreamStatus(req *api.StreamStatusRequest, svc api.RolloutSe
...
+	}
+
 	return &api.GetStatusResponse{
 		Status:       status,
-		Applications: apps,
+		Applications: appList,
 	}, nil
 }
 
+// Removes irrelevant apps from
```
Comment (sven-urbanski-freiheit-com) [inline services/rollout-service/pkg/service/broadcast.go:254]: from what?

File: services/rollout-service/pkg/service/broadcast_test.go [hunk 3]
```diff
@@ -671,5 +765,33 @@ func TestGetStatus(t *testing.T) {
 			}
 		})
 
+		// This runs all test-cases again but delays all argoevents.
+		// The efect is that all apps will start as "unknown" and the will eventually converge.
```
Comment (sven-urbanski-freiheit-com) [inline services/rollout-service/pkg/service/broadcast_test.go:769]:
```suggestion
		// The effect is that all apps will start as "unknown" and then will eventually converge.
```

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: services/rollout-service/pkg/service/broadcast.go (hunk 1 above)
Comment (hannesg) [reply services/rollout-service/pkg/service/broadcast.go:214]: Yes, that is correct.
